### PR TITLE
Adding documentation for synthetic highk/dbs

### PR DESCRIPTION
**Pull request by juanruizruiz**
Work in progress.

### CODE REVIEW

File: docs/howtos/syn_hkdbs.rst [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+    thetatmp = self.geometry.theta[self.geometry.Z > self.geometry.Z0]
+    Rtmp = self.geometry.R[self.geometry.Z > self.geometry.Z0] * self.a_minor  # [m]
+    Ztmp = self.geometry.Z[self.geometry.Z > self.geometry.Z0] * self.a_minor  # [m]
+    tmp_ind = np.argmin(np.abs(Rtmp - Rloc))
+    self.thetaloc = thetatmp[tmp_ind]  # np.interp(Zloc, Ztmp, thetatmp)
+    self.Rtmp = Rtmp[tmp_ind]
+    self.Ztmp = Ztmp[tmp_ind]
+
+This gives the following plot: 
+.. image:: figures/GS2_mode_frequency_plot.png
```
Comment (bpatel2107) [inline docs/howtos/syn_hkdbs.rst]:
```suggestion
.. image:: figures/jet_example_scatloc.png       
```

This should be

File: docs/howtos/syn_hkdbs.rst [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+    self.Ztmp = Ztmp[tmp_ind]
+
+This gives the following plot: 
+.. image:: figures/GS2_mode_frequency_plot.png       
+   :width: 600
+
+3. Next, call the function mapk. Given a pair (kn, kb), we calculate the corresponding (kx, ky) in the simulation grid. Here, we need to first define a right handed coordinate system. 
+We use the basis of unit vectors :math:`(\hat{\mathbf{b}}, \hat{\mathbf{e}}_n, \hat{\mathbf{e}}_b)`. Here :math:`\hat{\mathbf{b}}` is along the background magnetic field. 
+The normal unit vector :math:`\hat{\mathbf{e}}_n = \nabla \psi/|\nabla \psi|` is normal to the flux surface. 
+The binormal unit vector :math:`\hat{\mathbf{e}}_b = \hat{\mathbf{b}} \times \hat{\mathbf{e}}_n ` is in the binormal direction, that is, in the flux surface and perpendicular to :math:`\hat{\mathbf{b}}`.
```
Comment (bpatel2107) [inline docs/howtos/syn_hkdbs.rst]:
```suggestion
The binormal unit vector :math:`\hat{\mathbf{e}}_b = \hat{\mathbf{b}} \times \hat{\mathbf{e}}_n` is in the binormal direction, that is, in the flux surface and perpendicular to :math:`\hat{\mathbf{b}}`.
```

Remove extra space

File: docs/howtos/syn_hkdbs.rst [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+        eq_file=eq_file,
+        kinetics_file=kinetics_file,
+        simdir=simdir,
+        savedir=simdir,
+        fsize=fsize,
+    )
+
+In the __init__ of SyntheticHighkDBS, we also calculate the scattering location in theta from a given (Rloc,Zloc) location. 
+For that, we first find the radial location (poloidal flux) corresponding to (Rloc, Zloc). 
+Then, we fit the flux surface using the local geometry specification (eq. "Miller", "MHX"), similar to what should be used in the GK simulation output.
```
Comment (bpatel2107) [inline docs/howtos/syn_hkdbs.rst]:
```suggestion
Then, we fit the flux surface using the local geometry specification (eq. "Miller", "MXH"), similar to what should be used in the GK simulation output. 
```

File: docs/howtos/syn_hkdbs.rst [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+The following discusses how one uses the synthetic diagnostic for ``pyrokinetics``. The steps in the synthetic diagnostic are the following:
+1. Inputs are diagnostic specific (diagnostic, filter, k, location, resolution, local rhos). See example_syn_hk_dbs.py
+2. Load equilibrium, kinetics files. Find scattering location theta. See __init__ in class SyntheticHighkDBS
+3. Map (kn, kb) to (kx, ky) for all k's / channels specified in 1. See function mapk
+4. Load GK output data (fluctuation moment file). See function get_syn_fspec 
+5. For each input condition (eg. for each k in highk/DBS), filter simulation data. See class Filter, functions apply_filter, get_syn_fspec
+6. Generate synthetic spectra and make plots. See functions get_syn_fspec, plot_syn
+
+
+1. We first point is to provide input data from the specific diagnostic (high-k, DBS for now). See example_syn_hk_dbs.py as reference.
```
Comment (bpatel2107) [inline docs/howtos/syn_hkdbs.rst]:
```suggestion
## 1. 
The first point is to provide input data from the specific diagnostic (high-k, DBS for now). See example_syn_hk_dbs.py as reference. 
```

File: docs/howtos/syn_hkdbs.rst [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+    wR = 0.1 * ureg.meter       #
+    wZ = 0.05 * ureg.meter      # 
+    eq_file = "path_to_equilibrium_file"
+    kinetics_file = eq_file
+    simdir = "path_to_simulation_directory"
+    savedir = "path_to_save_figures"
+    if_save = 0
+    fsize = 22
+
+2. Next, call SyntheticHighkDBS, which defines the syn_diag object.
```
Comment (bpatel2107) [inline docs/howtos/syn_hkdbs.rst]:
```suggestion
## 2. 
Next, call SyntheticHighkDBS, which defines the syn_diag object. 
```

File: docs/howtos/syn_hkdbs.rst [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+    tmp_ind = np.argmin(np.abs(Rtmp - Rloc))
+    self.thetaloc = thetatmp[tmp_ind]  # np.interp(Zloc, Ztmp, thetatmp)
+    self.Rtmp = Rtmp[tmp_ind]
+    self.Ztmp = Ztmp[tmp_ind]
+
+This gives the following plot: 
+.. image:: figures/GS2_mode_frequency_plot.png       
+   :width: 600
+
+3. Next, call the function mapk. Given a pair (kn, kb), we calculate the corresponding (kx, ky) in the simulation grid. Here, we need to first define a right handed coordinate system.
```
Comment (bpatel2107) [inline docs/howtos/syn_hkdbs.rst]:
```suggestion
## 3. 
Next, call the function mapk. Given a pair (kn, kb), we calculate the corresponding (kx, ky) in the simulation grid. Here, we need to first define a right handed coordinate system. 
```

File: docs/howtos/syn_hkdbs.rst [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+    \end{equation}
+
+where :math:`k_x = 2 \pi p / L_x` is the radial wave number definition in pyro, :math:`n` is the toroidal mode number, and :math:`L_x` is the radial extent of the numerical simulation. 
+The mapping in equation :math:`\ref{knkb_map}` is performed within the function mapk.py, and executed as follows: 
+
+.. code:: python
+    # map k
+    syn_diag.mapk()
+
+Next, apply the synthetic diagnostic. Use get_syn_fspec and plot_syn
```
Comment (bpatel2107) [inline docs/howtos/syn_hkdbs.rst]:
```suggestion
## 6. 
Next, apply the synthetic diagnostic. Use get_syn_fspec and plot_syn
```

File: docs/howtos/syn_hkdbs.rst [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+        Using synthetic diagnostic
+============================================
+
+The following discusses how one uses the synthetic diagnostic for ``pyrokinetics``. The steps in the synthetic diagnostic are the following:
+1. Inputs are diagnostic specific (diagnostic, filter, k, location, resolution, local rhos). See example_syn_hk_dbs.py
+2. Load equilibrium, kinetics files. Find scattering location theta. See __init__ in class SyntheticHighkDBS
+3. Map (kn, kb) to (kx, ky) for all k's / channels specified in 1. See function mapk
+4. Load GK output data (fluctuation moment file). See function get_syn_fspec 
+5. For each input condition (eg. for each k in highk/DBS), filter simulation data. See class Filter, functions apply_filter, get_syn_fspec
+6. Generate synthetic spectra and make plots. See functions get_syn_fspec, plot_syn
```
Comment (bpatel2107) [inline docs/howtos/syn_hkdbs.rst]:
```suggestion
The following discusses how one uses the synthetic diagnostic for ``pyrokinetics``. The steps in the synthetic diagnostic are the following:

1. Inputs are diagnostic specific (diagnostic, filter, k, location, resolution, local rhos). See example_syn_hk_dbs.py
2. Load equilibrium, kinetics files. Find scattering location theta. See __init__ in class SyntheticHighkDBS
3. Map (kn, kb) to (kx, ky) for all k's / channels specified in 1. See function mapk
4. Load GK output data (fluctuation moment file). See function get_syn_fspec 
5. For each input condition (eg. for each k in highk/DBS), filter simulation data. See class Filter, functions apply_filter, get_syn_fspec
6. Generate synthetic spectra and make plots. See functions get_syn_fspec, plot_syn
```

I think a space is needed to trigger the list

File: docs/howtos/syn_hkdbs.rst [hunk 9]
```diff
@@ -0,0 +1,145 @@
...
+    \begin{equation}
+        \begin{alignedat}{2}
+        & k_n = \mathbf{k}_\perp \cdot \hat{\mathbf{e}}_n = - n \frac{\nabla \alpha \cdot \nabla r}{| \nabla r |} + k_x |\nabla r|, \\
+        & k_b = \mathbf{k}_\perp \cdot \hat{\mathbf{e}}_b = - n \left( \hat{\mathbf{b}} \times \frac{\nabla r}{r} \right) \cdot \nabla \alpha
+        \end{alignedat}
+        \label{knkb_map}
+    \end{equation}
+
+where :math:`k_x = 2 \pi p / L_x` is the radial wave number definition in pyro, :math:`n` is the toroidal mode number, and :math:`L_x` is the radial extent of the numerical simulation. 
+The mapping in equation :math:`\ref{knkb_map}` is performed within the function mapk.py, and executed as follows:
```
Comment (bpatel2107) [inline docs/howtos/syn_hkdbs.rst]:
```suggestion
The mapping in equation \ref{knkb_map} is performed within the function mapk.py, and executed as follows: 
```

I think we need to remove the math directive

File: docs/howtos/syn_hkdbs.rst (hunk 4 above)
Comment (juanruizruiz) [reply docs/howtos/syn_hkdbs.rst]: I made a typo, 'The first point is ...' not 'We first point is ...'.

**Review by bpatel2107 — APPROVED** [review]
This looks fantastic! Made some small tweaks but happy for this to be merged

**Pull request merged**